src/linter/yaml/UI5YamlLinter.ts

```diff
@@ -0,0 +1,87 @@
+import {LintMessageSeverity} from "../LinterContext.js";
+import LinterContext from "../LinterContext.js";
+import deprecatedLibraries from "../../utils/deprecatedLibs.js";
+import yaml from "js-yaml";
+import {DataWithPosition, fromYaml, getPosition} from "data-with-position";
+
+// file content schema of 'UI5Yaml' with only relevant properties
+interface UI5YamlContentSchema { // extend for further detections
+	framework: {
+		libraries: {
+			name: string;
+		}[];
+	};
+}
+
+interface UI5YamlContentSchemaWithPosInfo extends DataWithPosition {
+	framework?: {
+		libraries?: {
+			name: string;
+		}[];
+	};
+	positionKey?: {
+		end: {
+			column: number;
+			line: number;
+		};
+		start: {
+			column: number;
+			line: number;
+		};
+	};
+}
+
+export default class UI5YamlLinter {
+	#content = "";
+	#yamlContentWithPosInfo: UI5YamlContentSchemaWithPosInfo = {};
```

Comment:
```suggestion
	#yamlContentWithPosInfo: UI5YamlContentSchemaWithPosInfo | undefined;
```

Please correct me if I'm missing something, but I think initializing this property with an empty object is misleading since that's not the true content of the yaml.

This makes it impossible to distinguish whether the yaml has been parsed but is empty, or hasn't been parsed yet. I would rather leave the property undefined until the yaml content is parsed.

Reply:
e52efc0